mbed_targets/mbed_targets.py

```diff
@@ -90,6 +91,26 @@ def from_offline_target_entry(cls, target_entry: dict) -> "MbedTarget":
             build_variant=tuple(target_entry.get("build_variant", [])),
         )
 
+    def fetch_target_build_attributes(self, path_to_targets_json: str) -> Any:
+        """When provided with path to a targets.json file in Mbed OS, returns a dict of attributes.
+
+        These attributes contains the specific information needed to build Mbed applications for this Mbed Target.
+
+        Args:
+            path_to_targets_json: path to a targets.json file found in the Mbed OS library
+
+        Returns:
+            A dict containing the parsed attributes from targets.json
+
+        Raises:
+            FileNotFoundError: path provided does not lead to targets.json
+            TargetBuildAttributesError: an error has occurred while fetching build attributes
+        """
+        try:
+            return target_attributes.get_target_attributes(path_to_targets_json, self.board_name)
+        except (FileNotFoundError, target_attributes.TargetAttributesError) as e:
+            raise TargetBuildAttributesError(e)
```

Comment:
Should we use `raise from` here so we preserve the full traceback?

Reply:
56383a8